ush/templates/FV3LAM_wflow.xml

```diff
@@ -295,15 +311,13 @@ MODULES_RUN_TASK_FP script.
     <envar><name>ICS_OR_LBCS</name><value>LBCS</value></envar>
 
     <dependency>
-      <!-- Files through fhr 39 exist. -->
-      <sh>
-        test $(find <cyclestr offset="-{{ extrn_mdl_lbcs_offset_hrs }}:00:00">{{ extrn_mdl_sysbasedir_lbcs }}/ \
-                    -name @y@j@H*[0-3][0-9] -mmin +5 | wc -l)</cyclestr> \
-        == 40
-      </sh>
+       <and>
+         {%- for h in range(extrn_mdl_lbcs_offset_hrs, fcst_len_hrs+extrn_mdl_lbcs_offset_hrs+1, bc_update_interval) %}
+         <datadep age="00:00:05:00"> <cyclestr offset="-{{ extrn_mdl_lbcs_offset_hrs }}:00:00">{{ extrn_mdl_sysbasedir_lbcs }}/@y@j@H000{{ "%03d" % h }}</cyclestr></datadep>
+         {%- endfor %}
+       </and>
```

Comment:
Long lists of data dependencies can be tough on the file system when running rocoto. If you can rely on a single `find` command, the performance will likely be better.

Reply:
I had a hard time to figure out how to using "find" command to count the right number for boundary process. Like,  I used [0-1][0-8] == 18 but it is actually missing 09-h forecast when count. I don't know to count the first 12-h forecast grib2 files. Only checking the last grib2 forecast file works on operational environment. But it doesn't work for our environment because the forecast files from the same forecast do not arrive on JET in time order. The 39-h forecast could arrive much earlier than others. That is why I have to list all of the grib2 files for the boundary process as depends. Right now, it lists all of the grib2 files needed for 12-h forecast (longest forecast). In the future, we may need to create two "get_extrn_lbcs" tasks, one for long forecast and one for short forecast.

Reply:
Does this dependency work for cold start forecasts?

Reply:
Yes.